tests/Service/CaseManagerTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Tests\Service;
+
+use App\Entity\CaseEntity;
+use App\Entity\Municipality;
+use App\Repository\CaseEntityRepository;
+use App\Service\CaseManager;
+use PHPUnit\Framework\TestCase;
+
+class CaseManagerTest extends TestCase
+{
+    private $mockCaseRepository;
+    private $caseManager;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->mockCaseRepository = $this->createMock(CaseEntityRepository::class);
+
+        $this->caseManager = new CaseManager($this->mockCaseRepository);
+    }
+
+    public function testGenerateCaseNumberFirstCase()
+    {
+        $mockMunicipality = $this->createMock(Municipality::class);
+
+        $this->mockCaseRepository
+            ->expects($this->once())
+            ->method('findLatestCaseByMunicipality')
+            ->with($mockMunicipality)
+            ->willReturn(null);
+
+        $date = new \DateTime();
+        $year = $date->format('Y');
+
+        $result = $this->caseManager->generateCaseNumber($mockMunicipality);
+
+        $this->assertSame($year.'-0001', $result);
+    }
+
+    public function testGenerateCaseNumberPreviousYear()
+    {
+        $mockCase = $this->createMock(CaseEntity::class);
+
+        $mockMunicipality = $this->createMock(Municipality::class);
+
+        $this->mockCaseRepository
+            ->expects($this->once())
+            ->method('findLatestCaseByMunicipality')
+            ->with($mockMunicipality)
+            ->willReturn($mockCase);
+
+        $mockDate = $this->createMock(\DateTime::class);
+
+        $mockCase
+            ->expects($this->once())
+            ->method('getCreatedAt')
+            ->willReturn($mockDate);
+
+        $date = new \DateTime();
+        $currentYear = $date->format('Y');
+
+        $previousYear = (int) $currentYear - 1;
```

Comment:
Consider using something like `$previousYear = $currentYear->sub(new DateInterval('P1Y'))´ instead.

Reply:
Using `sub()` to subtract a year for testing purposes instead of doing it via type casting.